### ISSUE TITLE
BUG: Rules that are renamed/removed not removed on reload 

### ISSUE THREAD

**Issue by jfryman** [StackStorm/st2#924]
I ended up changing the namespace for my CI/CD workflow... from a url of `cicd/github/events` to `cicd/events`. On a reload however, the old endpoint is still registered.

Should this be removed, as it is no longer a defined endpoint? It is what I would expect to happen.

![10_20_30_2_3000___rules_548a2a07dc837b438b07d8dd_general](https://cloud.githubusercontent.com/assets/20028/5509525/a7f4e46a-877e-11e4-9dd1-ecc630c77e36.png)

**Comment by Kami** [comment]
Looks like I missed this ticket.

To clarify it, this is a feature and not a bug. We update all the resources in the DB when registration script is called, but we never delete them - we only delete them when user explicitly uninstalls a pack.

**Comment by manasdk** [comment]
Looks like a feature. Close the issue?

**Comment by jfryman** [comment]
I don't like this as a feature. If I delete something from my infra as code repository, I expect it gone. How do I reconcile this?

**Comment by manasdk** [comment]
Yeah we are deviating from the infrastructure as code promise to some extent. With the DB and filesystem as 2 sources of truth we will need to add some code to reconcile the sources. 

It is an issue that we have brushed under the rug since there is a way to clean up old content.
